src/SonarQube.Client/SonarQubeService.cs

```diff
@@ -143,36 +148,26 @@ public async Task ConnectAsync(ConnectionInformation connection, CancellationTok
             logger.Info($"Connecting to '{connection.ServerUri}'.");
             logger.Debug($"IsConnected is {IsConnected}.");
 
-            httpClient = new HttpClient(messageHandler)
-            {
-                BaseAddress = connection.ServerUri, DefaultRequestHeaders = { Authorization = AuthenticationHeaderFactory.Create(connection.Credentials), },
-            };
-
-            httpClient.DefaultRequestHeaders.Add("User-Agent", userAgent);
-
             requestFactory = requestFactorySelector.Select(connection.IsSonarCloud, logger);
 
             try
             {
                 var serverTypeDescription = connection.IsSonarCloud ? "SonarCloud" : "SonarQube";
 
                 logger.Debug($"Getting the version of {serverTypeDescription}...");
-
-                var versionResponse = await InvokeUncheckedRequestAsync<IGetVersionRequest, string>(request => { }, token);
-                var serverInfo = new ServerInfo(Version.Parse(versionResponse), connection.IsSonarCloud ? ServerType.SonarCloud : ServerType.SonarQube);
+                var serverInfo = await GetServerInfo(connection, token);
 
                 logger.Info($"Connected to {serverTypeDescription} '{serverInfo.Version}'.");
+                currentHttpClient = CreateHttpClient(connection.ServerUri, connection.Credentials, ShouldUseBearer(serverInfo));
```

Comment:
not sure if `current` prefix is really needed here

Reply:
I like `current,` because it's consistent with the [currentServerInfo](https://github.com/SonarSource/sonarlint-visualstudio/blob/e65430e3a1b6719a347fe9a2695ee37cd1cce2c5/src/SonarQube.Client/SonarQubeService.cs#L45) field.
Additionally, inside [here](https://github.com/SonarSource/sonarlint-visualstudio/blob/e65430e3a1b6719a347fe9a2695ee37cd1cce2c5/src/SonarQube.Client/SonarQubeService.cs#L135), it will be easier to distinguish between the parameter and field